client/fake/fake_test.go

```diff
@@ -0,0 +1,1405 @@
+															"name":  {Type: "string"},
+															"image": {Type: "string"},
+														},
+														Required: []string{"name"},
+													},
+												},
+												XListType:    ptr.To("map"),
+												XListMapKeys: []string{"name"},
+											},
+										},
+									},
+								},
+							},
+						},
+					},
+				},
+			},
+		}
+
+		crdClient := NewFakeDynamicClientWithCRDs(scheme, []*apiextensionsv1.CustomResourceDefinition{crd})
+
+		appGVR := schema.GroupVersionResource{
+			Group:    "example.com",
+			Version:  "v1",
+			Resource: "myapps",
+		}
+
+		// Team A creates app with main container
+		appA := &unstructured.Unstructured{
+			Object: map[string]interface{}{
+				"apiVersion": "example.com/v1",
+				"kind":       "MyApp",
+				"metadata": map[string]interface{}{
+					"name":      "crd-strategic-merge-test",
+					"namespace": "default",
+				},
+				"spec": map[string]interface{}{
+					"containers": []interface{}{
+						map[string]interface{}{
+							"name":  "main-app",
+							"image": "nginx:1.21",
+						},
+					},
+				},
+			},
+		}
+
+		result1, err := crdClient.Resource(appGVR).Namespace("default").Apply(
+			ctx, "crd-strategic-merge-test", appA,
+			metav1.ApplyOptions{FieldManager: "team-a"},
+		)
+		require.NoError(t, err)
+
+		// Verify initial state
+		containers1, found, err := unstructured.NestedSlice(result1.Object, "spec", "containers")
+		require.NoError(t, err)
+		require.True(t, found)
+		require.Len(t, containers1, 1)
+
+		// Team B adds sidecar container - this should MERGE, not conflict
+		appB := &unstructured.Unstructured{
+			Object: map[string]interface{}{
+				"apiVersion": "example.com/v1",
+				"kind":       "MyApp",
+				"metadata": map[string]interface{}{
+					"name":      "crd-strategic-merge-test",
+					"namespace": "default",
+				},
+				"spec": map[string]interface{}{
+					"containers": []interface{}{
+						map[string]interface{}{
+							"name":  "sidecar",
+							"image": "envoy:1.18",
+						},
+					},
+				},
+			},
+		}
+
+		result2, err := crdClient.Resource(appGVR).Namespace("default").Apply(
+			ctx, "crd-strategic-merge-test", appB,
+			metav1.ApplyOptions{FieldManager: "team-b"},
+		)
+		require.NoError(t, err, "CRD strategic merge patch should work without conflicts")
+
+		containers2, found, err := unstructured.NestedSlice(result2.Object, "spec", "containers")
+		require.NoError(t, err)
+		require.True(t, found)
+
+		require.Len(t, containers2, 2, "Both containers should be present after strategic merge")
+
+		// Verify both containers are present
+		containerNames := make([]string, 0, 2)
+		for _, container := range containers2 {
+			containerMap := container.(map[string]interface{})
+			name := containerMap["name"].(string)
+			containerNames = append(containerNames, name)
+		}
+		require.Contains(t, containerNames, "main-app")
+		require.Contains(t, containerNames, "sidecar")
+	})
+}
```

Comment:
related to the other comment: if that's not the intended behaviour, adding a test here might help verify that the dry run result shows the changes, but the actual object is unchanged.